app/src/main/java/com/duckduckgo/app/trackerdetection/model/NetworkTrackers.kt

```diff
@@ -37,4 +57,13 @@ class NetworkTrackers @Inject constructor() : Serializable {
                 .map { it.networkName }
                 .firstOrNull()
     }
+
+    fun majorNetwork(url: String): String? {
+        return majorNetworks
+                .filter { network(url).equals(it.name, true) }
+                .map { it.name }
+                .firstOrNull()
```

Comment:
This could probably be simpler and more efficient using a Kotlin array for majorNetworks and then use `.find` , rather than filtering -> map -> find first or null, e.g.  line 34 could be:

    majorNetworks = arrayOf(

then line 62 could be:

    return majorNetworks.find { network(url).equals(it.name, true)  }?.name

see https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/find.html